packages/p2p/src/components/my-profile/payment-methods/payment-method-card/payment-method-card.jsx

```diff
@@ -22,16 +24,29 @@ const PaymentMethodCard = ({
     style,
 }) => {
     const { general_store, my_ads_store, my_profile_store } = useStores();
-    const method = !is_add && payment_method?.display_name.replace(/\s|-/gm, '');
-    const payment_account = payment_method?.fields?.account?.value;
-    const payment_account_name = payment_method?.display_name;
-    const payment_bank_name = payment_method?.fields?.bank_name?.value;
-    const payment_name = payment_method?.fields?.name?.value;
-    const payment_method_name = payment_method?.display_name.replace(/\s|-/gm, '');
-    const icon_method =
-        payment_method_name === 'BankTransfer' || payment_method_name === 'Other'
-            ? `IcCashier${payment_method_name}`
-            : 'IcCashierEwallet';
+    const { showModal } = useModalManagerContext();
+    const { display_name, fields, icon, id } = payment_method || {};
+    const method = !is_add && display_name.replace(/\s|-/gm, '');
+    const payment_account = fields?.account?.value;
+    const payment_account_name = display_name;
+    const payment_bank_name = fields?.bank_name?.value;
+    const payment_name = fields?.name?.value;
+
+    const handleEditDeletPaymentMethod = e => {
+        if (e.target.value === 'delete') {
+            showModal({
+                key: 'DeletePaymentMethodConfirmationModal',
+                props: {
+                    payment_method_id: id,
+                    payment_method_name: fields?.bank_name?.value || fields?.name?.value || display_name,
```

Comment:
would this work? if not can ignore this comment

```suggestion
                    payment_method_name: payment_bank_name || payment_name || display_name,
```